gcn/main.py

```diff
@@ -0,0 +1,121 @@
+from argparse import ArgumentParser
+
+import mlx.core as mx
+import mlx.nn as nn
+import mlx.optimizers as optim
+from mlx.nn.losses import cross_entropy
+from mlx.utils import tree_flatten
+
+from datasets import download_cora, load_data, train_val_test_mask
+from gcn import GCN
+
+
+def loss_fn(y_hat, y, weight_decay=0.0, parameters=None):
+    l = mx.mean(nn.losses.cross_entropy(y_hat, y))
+
+    if weight_decay != 0.0:
+        assert parameters != None, "Model parameters missing for L2 reg."
+
+        l2_reg = mx.zeros(
+            1,
+        )
```

Comment:
This formatting looks a little off, make sure you run `black` (see e.g. the [CONTRIBUTIN.md](https://github.com/ml-explore/mlx-examples/blob/main/CONTRIBUTING.md) file on the pre-commit hooks for more info)